Runtime/Async/MainThreadDispatcher/MainThreadDispatcherBase.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+
+namespace StansAssets.Foundation.Async
+{
+    abstract class MainThreadDispatcherBase : IMainThreadDispatcher
+    {
+        static readonly Queue<Action> s_ExecutionQueue = new Queue<Action>();
```

Comment:
I would suggest using `ConcurrentQueue<Action>` instead of usual `Queue<Action>` with a manual `lock` to have thread-safe FIFO collection right from the box.